### PR TITLE
[FIX] 0036105: mob- and fileupload field data type - ILIAS don`t save the input

### PR DESCRIPTION
**Pull request by chfsx**
Hi @Amstutz 
This is a fix for https://mantis.ilias.de/view.php?id=36105, I hope i tackled all situations where this happened

### CODE REVIEW

**Comment by chfsx** [comment]
Failing UniTests are due to a problem with PHPUnit-version which is used in github-runs.

File: Modules/DataCollection/classes/Content/class.ilDclRecordEditGUI.php [hunk 1]
```diff
@@ -82,6 +82,63 @@ public function __construct(ilObjDataCollectionGUI $parent_obj, int $table_id, i
         $this->tableview = ilDclTableView::findOrGetInstance($this->tableview_id);
     }
 
+    protected function rebuildUploadsFroFileHash(bool $has_ilfilehash): string
```
Comment (Amstutz) [inline Modules/DataCollection/classes/Content/class.ilDclRecordEditGUI.php]: minor typo, should be For, right? Or Form?

**Review by Amstutz — CHANGES_REQUESTED** [review]
Thx you soo much. Special thx for factoring out the build hash function. makes the code much more readable. Other minor improvements are also highly appreciated.

There is just one minor typo (see inline comment). I would fix it myself, however, I am not a 100% sure, if it should be For or From. You may directly merge afterwards, if you pick to trunk as well. You can also pass this back to me for picking to trunk. whatever works best for you.

Thx a lot!

**Comment by chfsx** [comment]
Nice catch, thanks! I'll merge this and CP to trunk as well.

**Comment by chfsx** [comment]
@Amstutz cherry-picking to trunk is currently not possible, trunk seems to have a quite different stage, there are differences between release_8 and trunk in at least this files:
```
Modules/DataCollection/classes/Adapters/Ui/class.ilDataCollectionUiAdapter.php
Modules/DataCollection/classes/Content/class.ilDclRecordEditGUI.php
Modules/DataCollection/classes/CreateView/class.ilDclCreateViewDefinitionGUI.php
Modules/DataCollection/classes/EditView/class.ilDclEditViewDefinitionGUI.php
Modules/DataCollection/classes/Fields/Base/class.ilDclBaseFieldModel.php
Modules/DataCollection/classes/Fields/Base/class.ilDclBaseFieldRepresentation.php
Modules/DataCollection/classes/Fields/Base/class.ilDclBaseRecordFieldModel.php
Modules/DataCollection/classes/Fields/Base/class.ilDclBaseRecordModel.php
Modules/DataCollection/classes/Fields/Base/class.ilDclStandardField.php
Modules/DataCollection/classes/Fields/Boolean/class.ilDclBooleanFieldRepresentation.php
Modules/DataCollection/classes/Fields/Datetime/class.ilDclDatetimeFieldRepresentation.php
Modules/DataCollection/classes/Fields/Fileupload/class.ilDclFileuploadFieldRepresentation.php
Modules/DataCollection/classes/Fields/Fileupload/class.ilDclFileuploadRecordFieldModel.php
Modules/DataCollection/classes/Fields/Formula/class.ilDclFormulaFieldRepresentation.php
Modules/DataCollection/classes/Fields/IliasReference/class.ilDclIliasReferenceFieldRepresentation.php
Modules/DataCollection/classes/Fields/IliasReference/class.ilDclIliasReferenceRecordFieldModel.php
Modules/DataCollection/classes/Fields/Mob/class.ilDclMobFieldRepresentation.php
Modules/DataCollection/classes/Fields/Mob/class.ilDclMobRecordRepresentation.php
Modules/DataCollection/classes/Fields/Number/class.ilDclNumberFieldRepresentation.php
Modules/DataCollection/classes/Fields/Rating/class.ilDclRatingFieldRepresentation.php
Modules/DataCollection/classes/Fields/Selection/class.ilDclSelectionFieldRepresentation.php
Modules/DataCollection/classes/Fields/Text/class.ilDclTextFieldRepresentation.php
Modules/DataCollection/classes/Fields/class.ilDclFieldEditGUI.php
Modules/DataCollection/classes/Fields/class.ilDclFieldFactory.php
Modules/DataCollection/classes/Helpers/class.ilDclCache.php
Modules/DataCollection/classes/Helpers/class.ilDclPropertyFormGUI.php
Modules/DataCollection/classes/Helpers/class.ilDclTextInputGUI.php
Modules/DataCollection/classes/Table/class.ilDclTable.php
Modules/DataCollection/classes/TableView/class.ilDclTableView.php
Modules/DataCollection/classes/TableView/class.ilDclTableViewBaseDefaultValue.php
Modules/DataCollection/classes/TableView/class.ilDclTableViewFieldSetting.php
Modules/DataCollection/classes/TableView/class.ilDclTableViewTableGUI.php
Modules/DataCollection/classes/class.ilObjDataCollection.php
Modules/DataCollection/classes/class.ilObjDataCollectionAccess.php
```
I suggest to update trunk first, since I assume trunk should have the same stage as release_8 at the moment since there aren't any new features on it, right? shall I make a PR for this?

**Comment by Amstutz** [comment]
I run into such issues before. Up to this point I just resolved the manually, to not overrite a possible new dev for 9 (which I am not aware of). However, since there are that many, I looked into it. Seems like this here was merged somehow, but not picked to trunk: https://github.com/ILIAS-eLearning/ILIAS/pull/5280/files . 

> I suggest to update trunk first, since I assume trunk should have the same stage as release_8 at the moment since there aren't any new features on it, right? shall I make a PR for this?

Yes, this is the best way. If there was a new feature at some point, the featuer. in question would be to be added again. I heavily doupt that this is the case. Like this we have a clean slate to move forward.

**Comment by chfsx** [comment]
I cherry-picked the changes now to trunk as well

**Pull request merged**